vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/json/JnMembersScalarEvaluator.java

```diff
@@ -34,68 +34,50 @@
 
 public class JnMembersScalarEvaluator extends AbstractTaggedValueArgumentScalarEvaluator {
     protected final IHyracksTaskContext ctx;
-    private final SequencePointable sp1, sp2;
+    private final SequencePointable sp;
     private final ArrayBackedValueStorage abvs;
     private final SequenceBuilder sb;
     private ArrayPointable ap;
+    private TaggedValuePointable tempTvp;
 
     public JnMembersScalarEvaluator(IHyracksTaskContext ctx, IScalarEvaluator[] args) {
         super(args);
         this.ctx = ctx;
-        sp1 = (SequencePointable) SequencePointable.FACTORY.createPointable();
-        sp2 = (SequencePointable) SequencePointable.FACTORY.createPointable();
+        sp = (SequencePointable) SequencePointable.FACTORY.createPointable();
         abvs = new ArrayBackedValueStorage();
         sb = new SequenceBuilder();
         ap = (ArrayPointable) ArrayPointable.FACTORY.createPointable();
+        tempTvp = (TaggedValuePointable) TaggedValuePointable.FACTORY.createPointable();
     }
 
     @Override
     protected void evaluate(TaggedValuePointable[] args, IPointable result) throws SystemException {
         TaggedValuePointable tvp = args[0];
-        TaggedValuePointable tvp1 = (TaggedValuePointable) TaggedValuePointable.FACTORY.createPointable();
         abvs.reset();
         sb.reset(abvs);
-        if (tvp.getTag() == ValueTag.SEQUENCE_TAG) {
-            TaggedValuePointable tempTvp = ppool.takeOne(TaggedValuePointable.class);
-            try {
-                tvp.getValue(sp1);
-                int size1 = sp1.getEntryCount();
+        try {
+            if (tvp.getTag() == ValueTag.SEQUENCE_TAG) {
+                tvp.getValue(sp);
+                int size1 = sp.getEntryCount();
                 for (int i = 0; i < size1; i++) {
-                    sp1.getEntry(i, tempTvp);
+                    sp.getEntry(i, tempTvp);
                     if (tempTvp.getTag() == ValueTag.ARRAY_TAG) {
-                        membersSequence(tempTvp, result, tvp1);
+                        tempTvp.getValue(ap);
+                        ap.appendSequence(sb);
                     } else {
                         XDMConstants.setEmptySequence(result);
```

Comment:
I don't think this is correct. Setting the empty sequence here is useless. When the sequence is saved to the result this value is overwritten. I think the else can be removed.